### PR TITLE
Add ability to check for update

### PR DESCRIPTION
**Pull request by JoelAtDeluxe**
This PR adds the ability to notify the user of new releases. This will not automatically install a new version (nor prevent access for an out-of-date version), but simply alert the user when one is available. 

Depends on: PR #8 (built off of this branch)

Note: this is "complete" and ready for review, except that we currently do not track versions for ATerm. A separate PR providing those values needs to be merged first (see PR #9 )

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by jkennedyvz — COMMENTED** [review]
Hi Joel,

This looks great. I'm holding off on approval until we get the other PRs closed.

**Comment by jrozner** [comment]
Does this handle untagged versions and what happens when the only version information we have is a hash?

**Comment by JoelAtDeluxe** [comment]
Regarding untagged versions and hashing:

The short version is that, currently, we can only act on proper semantic tags. 
In a situation where we lack a proper tag (either malformed, or completely missing), the system will default to a semantic version of `0.0.0-development`. In this situation, all updates are ignored, but the system chugs along fine.

File: cmd/aterm/climain.go [hunk 1]
```diff
@@ -51,6 +51,9 @@ func main() {
 		return
 	}
 
+	// TOOD: replace version, owner, repo with real values
+	appdialogs.NotifyUpdate("v30.0.0", "google", "go-github")
```
Comment (jrozner) [inline cmd/aterm/climain.go:55]: Code looks fine. Let's get #8 in and pull this data from the variables exposed by go-info and maybe setup a variable for the org/repo that we can populate from CI to automatically fill it in at build time like we do in the ashirt repo.

**Review by jrozner — CHANGES_REQUESTED** [review]

**Comment by JoelAtDeluxe** [comment]
This was based off of js-revised-start, which was squashed into master. This generated a lot of unnecessary conflicts to resolve, so instead migrated it to PR #15 (via git cherry-pick)

**Pull request closed without merging**